topostats/plotting.py

```diff
@@ -1,650 +1,362 @@
+            Variable to summarise.
+        molecule_id: str
+            Variable that uniquely identifies molecules.
+        image_id: str
+            Variable that uniquely identifies images.
+        hist: bool
+            Whether to plot histograms.
+        stat: str
+            Statistic to plot on histogram 'count' (default), 'freq'.
+        bins: int
+            Number of bins to plot.
+        kde: bool
+            Whether to include a Kernel Density Estimate.
+        cut: float = 20,
+            Cut point for KDE.
+        figsize: tuple
+            Figure dimensions.
+        alpha: float
+            Opacity to use in plots.
+        palette: str = "deep"
+            Seaborn colour plot to use.
+        xrange: tuple
+            Range of the x-axis, if too narrow will be over-ridden automatically to show the data.
+        file_ext: str
+            File type to save plots as 'png' (default), 'pdf', 'svg'.
+        output_dir: Union[str, Path]
+            Location to save plots to.
+        var_to_label: dict
+            Variable to label dictionary for automatically adding titles to plots.
+
+        Returns
+        =======
+        """
+        self.df = df if df is not None else pd.read_csv(csv_file)
+        self.stat_to_sum = stat_to_sum
+        self.molecule_id = molecule_id
+        self.image_id = image_id
+        self.hist = hist
+        self.bins = bins
+        self.stat = stat
+        self.kde = kde
+        self.cut = cut
+        self.figsize = figsize
+        self.alpha = alpha
+        self.palette = palette
+        self.xrange = xrange
+        self.file_ext = file_ext
+        self.output_dir = Path(output_dir)
+        self.output_dir.mkdir(parents=True, exist_ok=True)
+        self.var_to_label = var_to_label
+        self.melted_data = None
+        self.summary_data = None
+        self.label = None
+        self.sns_melt_data()
+        self.set_palette()
+        self._set_label(self.stat_to_sum)
+
+    def _setup_figure(self):
+        """Setup Matplotlib figure and axes."""
+        fig, ax = plt.subplots(1, 1, figsize=self.figsize)
+        return fig, ax
+
+    def _outfile(self, plot_suffix: str) -> str:
+        """Generate the output file name with the appropriate suffix.
+
+        Parameters
+        ----------
+        plot_suffix: str
+            The suffix to append to the output file.
+
+        Returns
+        -------
+        str:
+            Concanenated string of the outfile and plot_suffix.
+        """
+        return f"{self.stat_to_sum}_{plot_suffix}"
+
+    def sns_plot(self) -> None:
+        """Plot the distribution of one or more statistics as either histogram, kernel density estimates or both. Uses
+        base Seaborn."""
+        fig, ax = self._setup_figure()
+        if self.hist and not self.kde:
+            outfile = self._outfile("hist")
+            sns.histplot(data=self.melted_data, x="value", bins=self.bins, stat=self.stat, hue=self.image_id)
```

Comment:
When running `run_topostats` with default config and data, the histograms do not show bars, I tried editing the histogram 'bins' value between 5 and 200, and this did not fix it. I will investigate further once I have finished this review.

Reply:
Hmm, something has gone awry during the merging of `main`, they were working and I forgot to check.

The tests work as these are the test images (which pass locally, bet they're failing on CI!)

![image](https://user-images.githubusercontent.com/17566406/217571517-c90acf45-599f-4e45-8fbb-3f1d0693c0f1.png)

I'll investigate.

Reply:
Ahha! This is because of the SI units which are now all `e-08` or smaller and so the `xrange` (0-2) is valid, previously they were rescaled as values were outside of the user specified `0-2` range.

I've removed the ability to explicitly specify `xrange` I think its better to let this be determined from the data (which was done anyway if the range wasn't large enough).